src/sentry/api/endpoints/project_rules_configuration.py

```diff
@@ -64,7 +67,11 @@ def get(self, request: Request, project) -> Response:
             if rule_type.startswith("condition/"):
                 condition_list.append(context)
             elif rule_type.startswith("filter/"):
-                filter_list.append(context)
+                if (
+                    has_issue_severity_alerts
+                    or context["id"] != "sentry.rules.filters.issue_category.IssueSeverityFilter"
```

Comment:
@malwilley I don't understand this logic, shouldn't it be:

`has_issue_severity_alerts and context["id"] == "sentry.rules.filters.issue_category.IssueSeverityFilter"`

Reply:
The logic does work here - added a test to confirm [`5b822b6` (#55095)](https://github.com/getsentry/sentry/pull/55095/commits/5b822b62e3b079960e35d954f8b23e447308d22a)

This is in a loop over all filters, so we want the condition to pass for everything except `not has_issue_severity_alerts and context["id"] = "...IssueSeverityFilter"`